File: Models/Answer.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;

namespace a15.Models;

public class Answer {
    [Key]
    public int Id { get; set; }
    public string? Sql { get; set; }
    public DateTimeOffset? Timestamp { get; set; }
    public bool IsCorrect { get; set; }

    [ForeignKey("AttemptId")]
    public int? AttemptId { get; set; }
    [JsonIgnore]
    public Attempt? Attempt { get; set; } = null!;

    public int QuestionId { get; set; }
    public Question? Question { get; set; } = null!; // Navigation property

}

File: frontend/src/app/models/answer.ts
import * as _ from 'lodash-es';

export class Answer {
  id!: number;
  sql!: string;
  timestamp?: Date;
  isCorrect?: boolean;
  questionId?: number;
  attemptId?: number;

}

export interface AnswerDTO {
  id: number;
  sql: string;
  isCorrect: boolean;
  questionId: number;
}

  